sockeye/model.py

```diff
@@ -154,26 +166,66 @@ def save_version(folder: str):
         with open(fname, "w") as out:
             out.write(__version__)
 
+    def _get_embed_weights(self) -> Tuple[mx.sym.Symbol, mx.sym.Symbol, mx.sym.Symbol]:
+        """
+        Returns embedding parameters for source and target.
+
+        :return: Tuple of source and target parameter symbols.
+        """
+        assert isinstance(self.config.config_embed_source, encoder.EmbeddingConfig)
+        assert isinstance(self.config.config_embed_target, encoder.EmbeddingConfig)
+        w_embed_source = mx.sym.Variable(C.SOURCE_EMBEDDING_PREFIX + "weight",
```

Comment:
Put that into an if-clause for shared weights. I know it would not really make a difference, but it makes the code clearer.

Reply:
not sure I know what you mean.
Its much easier with overwriting the references below.

Reply:
I am not a big fan of (relatively complex) var initializations and then overwriting them. I would prefer something like

```python
if not self.config.weight_tying:
  w_embed_source = ...
  w_embed_target = ...
  w_out_target = ...
else:
  [...]
```
Makes it clearer, for me at least.

Reply:
Hmm, looking at the code in more detail I see that they are overwritten conditionally! Forget about this, then.